code/game/objects/effects/overlays.dm

```diff
@@ -25,18 +25,17 @@
 	mouse_opacity = 0
 	var/tmp/atom/BeamSource
 
-/obj/effect/overlay/beam/New(turf/loc, var/lifetime = 10)
+/obj/effect/overlay/beam/New(turf/loc, var/lifetime = 10, var/fade = 0)
 	..()
-	spawn()
-		sleep(lifetime/4)
-		alpha = 192
-		sleep(lifetime/4)
-		alpha = 128
-		sleep(lifetime/4)
-		alpha = 64
-		sleep(lifetime/4)
+	spawn if(fade)

```

Comment:
It might just be better in this case to do
if(fade)
 spawn
  animate()
  qdel
else
 spawn(lifetime)
  qdel